packages/protocol/test-sol/unit/governance/validators/Validators.t.sol

```diff
@@ -804,6 +821,191 @@ contract ValidatorsTest_RegisterValidator is ValidatorsTest {
+  }
+
+  function test_ShouldSetValidatorSigner_WhenAccountHasAuthorizedValidatorSigner() public {
+    _whenL2();
+    _registerValidatorWithSignerHelper_noBls();
+    (, , , , address ActualSigner) = validators.getValidator(validator);
+
+    assertEq(ActualSigner, signer);
+  }
+
+  function test_ShouldSetLockGoldRequirements_WhenAccountHasAuthorizedValidatorSigner() public {
+    _whenL2();
+    _registerValidatorWithSignerHelper_noBls();
+    uint256 _lockedGoldReq = validators.getAccountLockedGoldRequirement(validator);
+
+    assertEq(_lockedGoldReq, originalValidatorLockedGoldRequirements.value);
+  }
+
+  function test_ShouldSetValidatorMembershipHistory_WhenAccountHasAuthorizedValidatorSigner()
+    public
+  {
+    _whenL2();
+    _registerValidatorWithSignerHelper_noBls();
+    (uint256[] memory _epoch, address[] memory _membershipGroups, , ) = validators
+      .getMembershipHistory(validator);
+
+    uint256[] memory validatorRegistrationEpochNumberList = new uint256[](1);
+    validatorRegistrationEpochNumberList[0] = validatorRegistrationEpochNumber;
+    address[] memory expectedMembershipGroups = new address[](1);
+    expectedMembershipGroups[0] = address(0);
+
+    assertEq(_epoch, validatorRegistrationEpochNumberList);
+    assertEq(_membershipGroups, expectedMembershipGroups);
+  }
+
+  function testFail_DoesNotEmit_ValidatorBlsPublicKeyUpdatedEvent() public {
+    _whenL2();
+    (bytes memory _ecdsaPubKey, uint8 v, bytes32 r, bytes32 s) = _generateEcdsaPubKeyWithSigner(
+      validator,
+      signerPk
+    );
+
+    vm.prank(validator);
+    accounts.authorizeValidatorSigner(signer, v, r, s);
+
+    vm.expectEmit(true, true, true, true);
+    emit ValidatorBlsPublicKeyUpdated(validator, blsPublicKey);
+
+    vm.prank(validator);
+    validators.registerValidator(_ecdsaPubKey);
+  }
+
+  function test_Emits_ValidatorRegisteredEvent() public {
+    _whenL2();
+    (bytes memory _ecdsaPubKey, uint8 v, bytes32 r, bytes32 s) = _generateEcdsaPubKeyWithSigner(
+      validator,
+      signerPk
+    );
+
+    vm.prank(validator);
+    accounts.authorizeValidatorSigner(signer, v, r, s);
+
+    vm.expectEmit(true, true, true, true);
+    emit ValidatorRegistered(validator);
+
+    vm.prank(validator);
+    validators.registerValidator(_ecdsaPubKey);
+  }
+
+  function test_Reverts_WhenAccountAlreadyRegisteredAsValidator() public {
+    _whenL2();
+    bytes memory _registeredEcdsaPubKey = _registerValidatorWithSignerHelper_noBls();
+    vm.expectRevert("Already registered");
+    vm.prank(validator);
+    validators.registerValidator(_registeredEcdsaPubKey);
+  }
+
+  function test_Reverts_WhenAccountAlreadyRegisteredAsValidatorGroup() public {
+    _whenL2();
+    _registerValidatorGroupHelper(validator, 1);
+    vm.expectRevert("Already registered");
+    vm.prank(validator);
+    validators.registerValidator(
+      abi.encodePacked(bytes32(0x0101010101010101010101010101010101010101010101010101010101010101))
```

Comment:
Any reason for not using the `_generateEcdsaPubKey` here to get the ecdsaPubKey?

Reply:
I mostly just copy-pasted the existing tests for the existing, CEL1 version of the function, that's how this test was implemented there.